### DISCUSSION TITLE
Mac OS Patch-Release Check

### DISCUSSION THREAD

**Discussion by jone** [ParetoSecurity/pareto-mac#84] [Ideas]
Hi

We have no check at the moment checking that the the Mac OS is up to date. I think this is very important and should be checked too.

In our company, most devs are still running on Big Sur and did not yet update to Monterey. I would not want everybody to update as soon as a new major is released, because usually some development problems need to be fixed first.

That's why I suggest these rules in the check:

1. You need to have the newest patch-release of your major version installed.
2. You don't need to have the newest major (e.g. Monterey), but one one of the still supported majors (see https://computing.cs.cmu.edu/desktop/os-lifecycle for example; I didn't dig down how that works in Mac OS though).

Example how to update a patch-release when not running the latest major, without upgrading to the next major:

<img width="875" alt="Bildschirmfoto 2021-12-08 um 09 25 08" src="https://user-images.githubusercontent.com/7469/145174193-8c882c6a-c221-4a00-8b66-42477ff008b8.png">

<img width="875" alt="Bildschirmfoto 2021-12-08 um 09 25 33" src="https://user-images.githubusercontent.com/7469/145174219-45c1c520-72c3-423d-b348-b91f3e420cf0.png">

**Comment by zupo** [comment]
Couldn't agree more, we're definitely doing this. 

The main thing I am still figuring out is how many major releases to consider "safe".

**Comment by buchi** [comment]
There seems to be no official policy from Apple about supporting macOS releases with security updates. In the past, the last three major versions of macOS usually received security updates. See: https://endoflife.date/macos. So I would propose to consider the last three major releases as "safe".

However with older hardware that doesn't support newer macOS releases, the latest supported release should still be considered ok.
This isn't a thing now, as Big Sur is required by the app anyway, if I'm not wrong.

**Reply by zupo** [reply]
Oh nice, that's good enough for me! 

That said, I disagree with the part about the older hardware. If OS is out-of-service, it has to be marked as so. The team admin can then disable the check if they want, but it must be clear that the device is using an OS version that no longer receives security updates.
